openmdao/core/component.py

```diff
@@ -726,6 +728,9 @@ def _declare_partials(self, of, wrt, dependent=True, rows=None, cols=None, val=N
             Value of subjacobian.  If rows and cols are not None, this will
             contain the values found at each (row, col) location in the subjac.
         """
+        if not dependent:
```

Comment:
Does this mean we can't do:

    self.declare_partials(of=['*'], wrt=['*'])
    self.declare_partials(of=['x'], wrt=['y'], dependent = False)

Or were we never able to do that even before. Plus I guess we have pursued the philosophy of avoiding '*' and always being explicit by listing all names, even though nobody does this in practce.

Reply:
OK I added a fix to make sure the scenario you mentioned still works.